pygmt/clib/session.py

```diff
@@ -1609,6 +1609,9 @@ def virtualfile_from_data(  # noqa: PLR0912
 
         return file_context
 
+    # virtualfile_from_data was renamed to virtualfile_in since v0.12.0.
+    virtualfile_from_data = virtualfile_in
```

Comment:
_Originally posted by @yvonnefroehlich in https://github.com/GenericMappingTools/pygmt/issues/3180#issuecomment-2088481434_

> Regarding the point
> ```
> * clib: Rename the "virtualfile_from_data" method to "virtualfile_in" ([#3068](https://github.com/GenericMappingTools/pygmt/pull/3068))
> ```
> I was / am a bit unsure, as there ins no FutureWarning added and not version stated when `virtualfile_from_data` is removed.

@seisman, did you want to keep this `virtualfile_from_data` alias around, or remove it at some point? If removing, should we start to raise a FutureWarning in v0.13.0, and remove it completely in v0.15.0?

Reply:
> If removing, should we start to raise a FutureWarning in v0.13.0, and remove it completely in v0.15.0?

Sounds good